opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java

```diff
@@ -258,74 +202,174 @@ public LoginResponse urlCanBeAccessWithGivenCredentials(String requestURL, Strin
     }
 
     public DownloadStatus downloadFromUrl(String url, String filename) {
-        setCredentials(allSharedPreferences.fetchRegisteredANM(), settings.fetchANMPassword());
-        Response<DownloadStatus> status = DownloadForm.DownloadFromURL(url, filename, httpClient);
+        Response<DownloadStatus> status = DownloadForm.downloadFromURL(url, filename,
+                allSharedPreferences.fetchRegisteredANM(), settings.fetchANMPassword());
         return status.payload();
     }
 
-    private void setCredentials(String userName, String password) {
-        httpClient.getCredentialsProvider()
-                .setCredentials(new AuthScope(configuration.host(), configuration.port(), REALM),
-                        new UsernamePasswordCredentials(userName, password));
+    public Response<String> fetchWithCredentials(String requestURL, String username, String password) {
+        HttpURLConnection urlConnection = null;
+        try {
+            urlConnection = initializeHttp(requestURL, false);
+
+            urlConnection.setRequestProperty("username", username);
+            urlConnection.setRequestProperty("password", password);
```

Comment:
Create  basic authorization header instead of this request property

Reply:
My understanding was that fetch uses basic auth and fetchWithCredentials uses username&password authentication. If they are the same, I will just call fetch in fetchWithCredentials

Reply:
the difference is fetch uses username and password saved in preferences or database while fetchWithCredentials they are passed explicitly.

fetchWithCredentials is important for call that needs to be made before login where the preferences/database is blank

Reply:
ok, understood.